packages/reactive-element/src/reactive-element.ts

```diff
@@ -1372,16 +1353,15 @@ export abstract class ReactiveElement
     debugLogEvent?.({kind: 'update'});
     if (!this.hasUpdated) {
       if (DEV_MODE) {
-        const shadowedProperties: string[] = [];
-        (
-          this.constructor as typeof ReactiveElement
-        ).__reactivePropertyKeys?.forEach((p) => {
-          if (this.hasOwnProperty(p) && !this.__instanceProperties?.has(p)) {
-            shadowedProperties.push(p as string);
-          }
-        });
+        // Produce warning if any reactive properties on the prototype are
+        // shadowed by class fields. Instance fields set before upgrade are
+        // deleted by this point, so any own property is caused by class field
+        // initialization in the constructor.
+        const ctor = this.constructor as typeof ReactiveElement;
+        const shadowedProperties = [...ctor.elementProperties.keys()].filter(
```

Comment:
Not exactly sure why this didn't use `elementProperties` previously, but I think it might have been to not warn on `noAccessor` properties, but that doesn't seem important given the prototype in check so this seems ok. Can double check with @bicknellr who I think did the initial warning code.

Reply:
I moved a `noAccessor` check to here before removing it in favor of the `in` check. The `in` check should subsume everything.